src/app/modules/form/public-form/public-form.controller.ts

```diff
@@ -294,65 +257,225 @@ export const handleGetPublicForm: ControllerHandler<
           })
           return new SgidMalformedMyInfoCookieError()
         },
-      )
-      return parseSgidMyInfoCookie()
-        .mapErr(() =>
-          res.json({
-            form: publicForm,
-            isIntranetUser,
-          }),
+      )()
+
+      if (parseSgidMyInfoCookieResult.isErr()) {
+        return res.json({
+          form: publicForm,
+          isIntranetUser,
+        })
+      }
+
+      const parseSgidMyInfoCookie = parseSgidMyInfoCookieResult.value
+      const { jwt: accessToken = '', sub = '' } = parseSgidMyInfoCookie
+
+      if (!accessToken) {
+        return res.json({
+          form: publicForm,
+          isIntranetUser,
+        })
+      }
+      res.clearCookie(SGID_MYINFO_COOKIE_NAME)
+      res.clearCookie(SGID_MYINFO_LOGIN_COOKIE_NAME)
+
+      const jwtPayloadResult =
+        await SgidService.extractSgidJwtMyInfoPayload(accessToken)
+      if (jwtPayloadResult.isErr()) {
+        const error = jwtPayloadResult.error
+        logger.error({
+          message: 'sgID: MyInfo login error',
+          meta: logMeta,
+          error,
+        })
+        return res.json({
+          form: publicForm,
+          myInfoError: true,
+          isIntranetUser,
+        })
+      }
+      const jwtPayload = jwtPayloadResult.value
+      const myInfoFieldsResult = await SgidService.retrieveUserInfo({
+        accessToken: jwtPayload.accessToken,
+        sub,
+      })
+
+      if (myInfoFieldsResult.isErr()) {
+        const error = myInfoFieldsResult.error
+        logger.error({
+          message: 'sgID: MyInfo login error',
+          meta: logMeta,
+          error,
+        })
+        return res.json({
+          form: publicForm,
+          myInfoError: true,
+          isIntranetUser,
+        })
+      }
+
+      myInfoFields = new SGIDMyInfoData(myInfoFieldsResult.value.data)
+      spcpSession = { userName: myInfoFields.getUinFin() }
+      break
+    }
+    default: {
+      return new UnreachableCaseError(authType)
```

Comment:
Mental note to remove use of `ts-essentials` and / or migrate to https://github.com/sindresorhus/type-fest -- both libraries are trying to do something similar.

Reply:
got it, i will not introduce any new inclusions to `ts-essentials` in the future, but will not change this for now to avoid unrelated scope for the PR.